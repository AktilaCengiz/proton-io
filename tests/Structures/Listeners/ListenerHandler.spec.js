const { EventEmitter } = require("events");
const Listener = require("../../../src/Structures/Listeners/Listener");
const ListenerHandler = require("../../../src/Structures/Listeners/ListenerHandler");
const ProtonClient = require("../../../src/Structures/ProtonClient");
const checkEqualKeys = require("../../checkEqualKeys");
const equal = require("../../equal");

const localEmitter = new EventEmitter();



class Client extends ProtonClient {
    constructor() {
        super({
            intents: []
        });

        this.listenerHandler = new ListenerHandler(this, {
            directory: `${process.cwd()}/tests/Structures/Listeners/ListenerModules`,
            automateCategories: true
        });

        this.listenerHandler.loadAll();

        this.listenerHandler.setEmitter("localEmitter", localEmitter);
    }
}

const client = new Client();

test("listen", () => {
    class Ready extends Listener {
        constructor() {
            super("Ready", {
                emitter: "localEmitter",
                type: "once",
                event: "ready"
            })
        }

        execute(message) {
            equal(message, "worked");
        }
    }

    client.listenerHandler.load(Ready);

    localEmitter.emit("ready", "worked");
})