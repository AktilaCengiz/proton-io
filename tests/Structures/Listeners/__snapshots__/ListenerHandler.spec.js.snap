// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`props 1`] = `
ListenerHandler {
  "_events": Object {},
  "_eventsCount": 0,
  "_maxListeners": undefined,
  "automateCategories": true,
  "client": Object {
    "application": null,
    "channels": Array [],
    "guilds": Array [],
    "owners": Array [],
    "presence": undefined,
    "shard": null,
    "user": null,
    "users": Array [],
  },
  "directory": "/home/heiwa/wioenena/workspace/packages/javascript/node/proton-io/tests/Structures/Listeners/ListenerModules",
  "emitters": Array [
    Object {
      "application": null,
      "channels": Array [],
      "guilds": Array [],
      "owners": Array [],
      "presence": undefined,
      "shard": null,
      "user": null,
      "users": Array [],
    },
    EventEmitter {
      "_events": Object {},
      "_eventsCount": 0,
      "_maxListeners": undefined,
      Symbol(kCapture): false,
    },
  ],
  "modules": Array [
    Object {
      "category": "ListenerModules",
      "client": Object {
        "application": null,
        "channels": Array [],
        "guilds": Array [],
        "owners": Array [],
        "presence": undefined,
        "shard": null,
        "user": null,
        "users": Array [],
      },
      "emitter": "client",
      "event": "ready",
      "execute": [Function],
      "filepath": "/home/heiwa/wioenena/workspace/packages/javascript/node/proton-io/tests/Structures/Listeners/ListenerModules/Basic.js",
      "handler": [Circular],
      "id": "Basic",
      "type": "once",
    },
  ],
  Symbol(kCapture): false,
}
`;
